source/include/FreeRTOS_IP.h

```diff
@@ -349,6 +349,9 @@ void FreeRTOS_SetEndPointConfiguration( const uint32_t * pulIPAddress,
 TaskHandle_t FreeRTOS_GetIPTaskHandle( void );
 
 void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
+                                     TickType_t uxBlockTimeTicks );
+
+void * FreeRTOS_GetUDPPayloadBuffer_ByIPType( size_t uxRequestedSizeBytes,
                                      TickType_t uxBlockTimeTicks,
                                      uint8_t ucIPType );
```

Comment:
Please add a parameter `ucIPType` to `FreeRTOS_GetUDPPayloadBuffer()`, and do not add a new function.

In labs/ipv6 we already had this function:
~~~
#if ( ipconfigUSE_IPv6 == 1 )
void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
                                     TickType_t uxBlockTimeTicks,
                                     uint8_t ucIPType )
~~~

Reply:
Hello Hein, Thanks for the review.

The reason for adding 2 functions is to maintain backward compatibility to projects that use the main branch. So the behavior  `FreeRTOS_GetUDPPayloadBuffer` is similar to main branch and the projects already using it can continue to use it without modifications, but `FreeRTOS_GetUDPPayloadBuffer_ByIPType` can be used for getting IPv6/IPv4 buffers.

Reply:
Ok, if you want to offer thius backward compatibility, please don't write the logic two times, but have

Reply:
Sure, I have made the changes. The comments for `FreeRTOS_GetUDPPayloadBuffer` have been updated to notify the user that its kept only for backward compatibility and the user should use `FreeRTOS_GetUDPPayloadBuffer_ByIPType` to allocate a IPv4 or IPv6 buffer.

Reply:
Sorry no, I prefer to keep `FreeRTOS_GetUDPPayloadBuffer()` only, along with the `ucIPType` parameter.
Normally I am also in favour of keeping functions backward compatible. But not in this case, because now the application writer must become aware that there are two types of UDP payload buffers: IPv4 and IPv6.

In labs/IPv6 we had this:

~~~
#if ( ipconfigUSE_IPv6 != 0 )
    void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
                                         TickType_t uxBlockTimeTicks,
                                         uint8_t ucIPType );
#else
    void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
                                         TickType_t uxBlockTimeTicks );
#endif
~~~

But as we stopped using the macro `ipconfigUSE_IPv6`, only the first function is left.

Reply:
Thanks Tony and Hein. I think it's best if we retain the original version of the API and create a new one for IPv4 and IPv6 scenario. The reason being : 1. We are following the same approach for the new API and we would like to be consistent 2. When the new code is merged with mainline, the existing IPv4 users will still be able to continue with the existing application.
However, I do agree that we can have a common logic in a single function and have 2 abstraction APIs to handle the 2 scenarios.